toolkit/tools/internal/isogenerator/isogenerator.go

```diff
@@ -0,0 +1,338 @@
+// Copyright Microsoft Corporation.
+// Licensed under the MIT License.
+
+package isogenerator
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+	"runtime"
+	"strings"
+
+	"github.com/cavaliercoder/go-cpio"
+	"github.com/klauspost/pgzip"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/file"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/logger"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safeloopback"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/safemount"
+	"github.com/microsoft/azurelinux/toolkit/tools/internal/shell"
+)
+
+const (
+	DefaultVolumeId                 = "CDROM"
+	efiBootImgPathRelativeToIsoRoot = "boot/grub2/efiboot.img"
+	initrdEFIBootDirectoryPath      = "boot/efi/EFI/BOOT"
+)
+
+type IsoGenConfig struct {
+	// Directory where temporary files can be stored.
+	BuildDirPath string
+	// Directory where to stage ISO files.
+	// If the directory exists, any existing files will be included in the ISO.
+	StagingDirPath string
+	// Enable legacy boot.
+	// Note: This isn't useful unless some additional assets are included in 'StagingDirPath'.
+	EnableBiosBoot bool
+	// The path where the ISO file will be written.
+	OutputFilePath string
```

Comment:
Chris, this is the output parameter, so maybe we keep it at the end?
- { build }
- { inputs }
- { outputs }

Reply:
Fixed.